### PR TITLE
add CLI for retrieving jupyter paths

### PR DESCRIPTION
**Pull request by minrk**
via `jupyter --config-dir`, etc.

Full paths can be retrieved as JSON with

    jupyter --paths --json

### CODE REVIEW

File: jupyter_core/command.py [hunk 1]
```diff
@@ -70,6 +83,37 @@ def main():
         parser = jupyter_parser()
         args, opts = parser.parse_known_args()
         subcommand = args.subcommand
+        if args.json and not args.paths:
+            print("--json is only used with --paths", file=sys.stderr)
+            sys.exit(1)
```
Comment (takluyver) [inline jupyter_core/command.py]: There's a handy shortcut for this - `sys.exit('string')` will print the string to stderr and exit with code 1.

File: jupyter_core/tests/test_command.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+"""Test the Jupiter
```
Comment (takluyver) [inline jupyter_core/tests/test_command.py]: Jupyter, despite your antipathy to the y ;-)

**Comment by takluyver** [comment]
A couple of minor comments, looks fine otherwise.

File: jupyter_core/tests/test_command.py (hunk 2 above)
Comment (minrk) [inline jupyter_core/tests/test_command.py]: GD autocorrect

File: jupyter_core/command.py (hunk 1 above)
Comment (minrk) [inline jupyter_core/command.py]: Nice.

**Comment by minrk** [comment]
Updated.

**Comment by Carreau** [comment]
+1

**Comment by rgbkrk** [comment]
Oh hey, this is the thing I asked about, for using with Hydrogen. I ended up doing both this and porting it to node. We went with the port.

File: jupyter_core/paths.py [hunk 3]
```diff
@@ -145,3 +145,27 @@ def jupyter_path(*subdirs):
...
+
+
+def jupyter_config_path():
+    """Return the search path for Jupyter config files as a list."""
+    paths = [jupyter_config_dir()]
+    for p in ENV_CONFIG_PATH:
+        if p not in SYSTEM_CONFIG_PATH:
+            paths.append(p)
+    paths.extend(SYSTEM_CONFIG_PATH)
+    return paths
```
Comment (rgbkrk) [inline jupyter_core/paths.py]: Good to know this is here too.

**Pull request merged**